### PR TITLE
#2705 - Enable new Program Year 2024-25

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
#### Program Year Selection

image

#### Part-time application NOA

image

#### Full-time application with parent

image

##### Supporting users providing info using the new form

image

##### Assessment ready after parent provided information using the new form

image

##### NOA Generated

image

### E2E Tests Required Refactors
- Dependents' eligibility was recently changed to Part-time and is now based on the offering starting date, which caused an issue when the new program year was added. Full-time eligibility is still based on the current day which requires the factory method to accommodate both.
- Tests related to 2021-2022 will be removed in a separate PR to avoid confusion. If added to the same PR many files will be treated as renamed.

### CODE REVIEW

File: sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-2023-2024-costs-child-care-cost.e2e-spec.ts
Comment (lewischen-aot) [inline sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-2023-2024-costs-child-care-cost.e2e-spec.ts:1]: I'm just wondering whether we should start removing the program years from the E2E test files. Also this file is under the 2024-2025 folder, but the file name has 2023-2024.

Comment (andrewsignori-aot) [reply sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-2023-2024-costs-child-care-cost.e2e-spec.ts:1]: As per the agreement we need to rename all the files under all program years and this is not part of the task.
_Duplicate the folder from sources\packages\backend\workflow\test\2023-2024 and rename it to 2024-2025. **Do not rename any program-year-specific named file**._
If the time allows I can think about doing one PR only for it.

Comment (lewischen-aot) [reply sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-2023-2024-costs-child-care-cost.e2e-spec.ts:1]: Ok. Got it.

**Review by lewischen-aot — APPROVED** [review]
Nice work setting up new program year 👍

**Review by sh16011993 — APPROVED** [review]
Thank You for enabling the new Program Year 2024-25 @andrewsignori-aot . LGTM 👍

**Pull request merged**